[YUNIKORN-1006] add notes about incompatible changes in the release doc (#61)
apache/incubator-yunikorn-release: docs/release-procedure.md

- Select the style `Text` and click `create`
- Scroll to the bottom of the page and copy the content of the text area and update the [CHANGELOG](../release-top-level-artifacts/CHANGELOG) file.

## Incompatible Changes

> YuniKorn community puts a high priority to the backwards compatibility. For each release, all incompatible

Reviewer: ```suggestion
YuniKorn community puts a high priority on backward compatibility. For each release, all incompatible
```

> changes must be carefully reviewed and documented. In general, incompatible changes to the public APIs usually requires

Reviewer: ```suggestion
changes must be carefully reviewed and documented. In general, incompatible changes to the public APIs usually require
```

> a major version bump, this needs to be discussed in the private list. Examples are REST API changes, scheduler-interface

Reviewer: ```suggestion
a major version bump and they need to be discussed in the private list. Examples are REST API changes, scheduler-interface
```

> main API changes. In the release notes, all incompatible changes must be clearly documented. User should be able to

Reviewer: ```suggestion
main API changes. In the release notes, all incompatible changes must be clearly documented. Users should be able to
```
